codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2PushPromiseFrameTest.java

```diff
@@ -0,0 +1,224 @@
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.util.CharsetUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.Assert.assertEquals;
+
+public class DefaultHttp2PushPromiseFrameTest {
+
+    private final EventLoopGroup eventLoopGroup = new NioEventLoopGroup(2);
+    private final ClientHandler clientHandler = new ClientHandler();
+    private final Map<Integer, String> contentMap = new ConcurrentHashMap<Integer, String>();
+
+    @Before
+    public void setup() throws InterruptedException {
+        ServerBootstrap serverBootstrap = new ServerBootstrap()
+                .group(eventLoopGroup)
+                .channel(NioServerSocketChannel.class)
+                .childHandler(new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    protected void initChannel(SocketChannel ch) {
+                        ChannelPipeline pipeline = ch.pipeline();
+
+                        Http2FrameCodec frameCodec = Http2FrameCodecBuilder.forServer()
+                                .autoAckSettingsFrame(true)
+                                .autoAckPingFrame(true)
+                                .build();
+
+                        pipeline.addLast(frameCodec);
+                        pipeline.addLast(new ServerHandler());
+                    }
+                });
+
+        ChannelFuture channelFuture = serverBootstrap.bind(0).sync();
+
+        final Bootstrap bootstrap = new Bootstrap()
+                .group(eventLoopGroup)
+                .channel(NioSocketChannel.class)
+                .handler(new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    protected void initChannel(SocketChannel ch) {
+                        ChannelPipeline pipeline = ch.pipeline();
+
+                        Http2FrameCodec frameCodec = Http2FrameCodecBuilder.forClient()
+                                .autoAckSettingsFrame(true)
+                                .autoAckPingFrame(true)
+                                .initialSettings(Http2Settings.defaultSettings().pushEnabled(true))
+                                .build();
+
+                        pipeline.addLast(frameCodec);
+                        pipeline.addLast(clientHandler);
+                    }
+                });
+
+        bootstrap.connect(channelFuture.channel().localAddress()).sync();
+    }
+
+    @Test
+    public void send() throws InterruptedException {
+        Thread.sleep(100);
```

Comment:
Seeing a sleep here seems like there needs to be something else that needs to be fixed